controller/server_status.py

```diff
@@ -292,6 +292,8 @@ def server_url_is_eventually_responsive(self, timeout_seconds: int = 5) -> bool:
     @property
     def overall_status(self) -> ServerStatusEnum:
         """Get the status of the jupyterserver."""
+        if self.hibernated:
```

Comment:
This exists in the following lines (line 306). Was it a reason to move it here? I believe the other two checks have higher priority than this one. Anyways we should remove the other one if this is required here.

Reply:
Ok I did not see that was already there. We should move it up because what Flora saw was that a session was failing because it was unschedulable. Then it was hibernated (in this unschedulable state) and the status went from failing to hibernated (as expected) but then after a few seconds it flipped back to failed/unschedulable. By moving this check up we avoid the problem. If the session is hibernated we should get the hibernated status regardless of what was going on when it was running. Especially now that we can modify the session when it is hibernated the old failing status is really not important.